features/theme/index.tsx

```diff
@@ -89,7 +89,9 @@ export function createTheme({
                 if (next.ignoreSystemMode) {
                     Colors.setScheme(scheme as SchemeType);
                 } else {
-                    Colors.setScheme('default');
+                    // Revert to system default if ignoreSystemMode is false
+                    const systemScheme = Appearance.getColorScheme() as 'light' | 'dark' | null;
+                    Colors.setScheme(systemScheme || 'default');
                 }
```

Comment:
Lines 89-95 are repeated down below on lines 126-131. You should make a function that takes `ignoreSystemMode` as a parameter and handle both with 1 function, instead of duplicating the logic in multiple places.